katello/backup.rb

```diff
@@ -28,10 +28,21 @@ def initialize(foreman_proxy_content, foreman_proxy, program, accepted_scenarios
       setup_opt_parser
     end
 
+    # do not use run_cmd() inside this method, to avoid a recursive loop
+    # as run_cmd() calls cleanup, and they will infinitely trigger one another
+    # rather, use system() call provided by ruby
     def cleanup(exitstatus=-1)
       puts "Cleaning up backup folder and starting any stopped services... "
       FileUtils.cd("/")
       FileUtils.rm_rf @dir unless @options[:no_subdir]
+      if @options[:snapshot]
+        @databases.each do |database|
+          mount_location = File.join(@mountdir, database)
+          system("umount #{mount_location}") if system("mount | grep #{mount_location}")
```

Comment:
instead of doing `mount | grep`, how about that function:

    def is_mounted(path)
      mounts = File.read('/proc/mounts')
      mounts.lines.each do |mount|
        mount = mount.split
        return true if mount[1] == path
      end
      return false
    end

Reply:
This seems much more costly and clunkier? I think it's also in the style guidelines to do avoid 'return true/false'.

Reply:
How about 
`if File.readlines('/proc/mounts').grep(/#{mount_location}).any?`